content/en/flux/installation/bootstrap/github.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Flux bootstrap for GitHub
+linkTitle: GitHub
+description: "How to bootstrap Flux with GitHub and GitHub Enterprise"
+weight: 20
+---
+
+The [flux bootstrap github](/flux/cmd/flux_bootstrap_github/) command deploys the Flux controllers
+on a Kubernetes cluster and configures the controllers to sync the cluster state from a GitHub repository.
+Besides installing the controllers, the bootstrap command pushes the Flux manifests to the GitHub repository
+and configures Flux to update itself from Git.
+
+After running the bootstrap command, any operation on the cluster (including Flux upgrades)
+can be done via Git push, without the need to connect to the Kubernetes cluster.
+
+{{% alert color="danger" title="Required permissions" %}}
+To bootstrap Flux, the person running the command must have **cluster admin rights** for the target Kubernetes cluster.
+It is also required that the person running the command to be the **owner** of the GitHub repository,
+or to have admin rights of a GitHub organization.
+{{% /alert %}}
+
+## GitHub PAT 
+
+For accessing the GitHub API, the boostrap command requires a GitHub personal access token (PAT)
+with administration permissions.
+
+The GitHub PAT can be exported as an environment variable:
+
+```sh
+export GITHUB_TOKEN=<gh-token>
+```
+
+If the `GITHUB_TOKEN` env var is not set, the bootstrap command will prompt you to type it the token.
+
+You can also supply the token using a pipe e.g. `echo "<gh-token>" | flux bootstrap github`.
+
+## GitHub Personal Account
+
+If you want to bootstrap Flux for a repository owned by a personal account, you can generate a
+[GitHub PAT](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line)
+that can create repositories by checking all permissions under `repo`.
+
+If you want to use an existing repository, the PAT's user must have `admin`
+[permissions](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role).
+
+Run the bootstrap for a repository on your personal GitHub account:
+
+```sh
+flux bootstrap github \
+  --token-auth \
+  --owner=my-github-username \
+  --repository=my-repository-name \
+  --branch=main \
+  --path=clusters/my-cluster \
+  --personal
+```
+
+If the specified repository does not exist, Flux will create it for you as private. If you wish to create
+a public repository, set `--private=false`.
+
+When using `--token-auth`, the CLI and the Flux controllers running on the cluster will use the GitHub PAT
```

Comment:
Speaking of Flux controllers, 

1. Do we also want to mention what these controllers are? Example: helm controller, kustomize controller, notification controller and source controller.
2. The need to add "— components-extra=image-reflector-controller,image-automation-controller" to the command if the image reflector controller and image automation controller are needed?

Source: https://medium.com/@aaltundemir/demystifying-gitops-bootstrapping-flux-fbfee94f8188

Reply:
All the bootstrap options should go under `Configuration`, once we have those, we can link from here.